src/main/java/org/springframework/integration/aws/inbound/kinesis/KinesisMessageDrivenChannelAdapter.java

```diff
@@ -784,6 +788,37 @@ public void run() {
 			};
 		}
 
+		private GetRecordsResult getRecords(GetRecordsRequest getRecordsRequest) {
+			try {
+				GetRecordsResult result = KinesisMessageDrivenChannelAdapter.this.amazonKinesis
+					.getRecords(getRecordsRequest);
+				return result;
+			}
+			catch (ExpiredIteratorException e) {
+				if (logger.isDebugEnabled()) {
+					logger.debug("Shard iterator " + getRecordsRequest.getShardIterator() + " expired");
+				}
+				// Iterator expired, but this does not mean that shard no longer contains records.
+				// Lets acquire iterator again (using checkpointer for iterator start sequence number).
+				this.state = ConsumerState.EXPIRED;
+			}
+			catch (ProvisionedThroughputExceededException e) {
+				if (logger.isDebugEnabled()) {
+					logger.debug("GetRecords request throttled for iterator " + getRecordsRequest.getShardIterator());
```

Comment:
According to its JavaDocs:
```
* The request rate for the stream is too high, or the requested data is too
 * large for the available throughput. Reduce the frequency or size of your
 * requests. 
```
it doesn't look like just to `sleep` is enough.
Let's make it like a `warn` and include the target `ProvisionedThroughputExceededException.message` to let end-user to know what was the reason of such an exception!